ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -257,6 +257,7 @@ primaryExpression
     | CASE whenClause+ (ELSE elseExpression=expression)? END                         #searchedCase
     | CAST '(' expression AS type ')'                                                #cast
     | ARRAY '[' (expression (',' expression)*)? ']'                                  #arrayConstructor
+    | '{' (identifier expression (',' identifier expression)*)? '}'                  #structConstructor
```

Comment:
Like the array constructor on the line above, I would propose something more along the lines of:

```suggestion
    | STRUCT '(' expression (AS? identifier)? (',' expression (AS? identifier))* ')'                  #structConstructor
```